src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadataViewImpl.java

```diff
@@ -20,125 +29,162 @@
+    @UiField
+    Div fileHistoryContainer;
+    @UiField
+    Span uploadDestinationPanel;
+    @UiField
+    Span uploadDestinationField;
+    @UiField
+    Text annotationsTitleText;
+    @UiField
+    ReactComponentDiv annotationsModalContainer;
+
+    @Inject
+    public EntityMetadataViewImpl(final SynapseJSNIUtils jsniUtils, final CookieProvider cookieProvider, final SynapseContextPropsProvider propsProvider) {
+        this.cookies = cookieProvider;
+        this.propsProvider = propsProvider;
+        initWidget(uiBinder.createAndBindUi(this));
+        idField.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                idField.selectAll();
+            }
+        });
+    }
+
+    @Override
+    public void setContainerItemCountWidget(IsWidget w) {
+        containerItemCountContainer.clear();
+        containerItemCountContainer.add(w);
+    }
+
+    @Override
+    public void setDoiWidget(IsWidget doiWidget) {
+        doiPanel.clear();
+        doiPanel.add(doiWidget);
+    }
+
+    @Override
+    public void setAnnotationsRendererWidget(IsWidget annotationsWidget) {
+        annotationsContainer.setWidget(annotationsWidget);
+    }
+
+    @Override
+    public void setUploadDestinationPanelVisible(boolean isVisible) {
+        uploadDestinationPanel.setVisible(isVisible);
+    }
+
+    @Override
+    public void setUploadDestinationText(String text) {
+        uploadDestinationField.setText(text);
+    }
+
+    @Override
+    public void setAnnotationsVisible(boolean visible) {
+        if (DisplayUtils.isInTestWebsite(cookies)) {
+            boolean showTabs = false;
+            if (visible) {
+                boolean showModal = true;
+                EntityModalProps props =
+                        EntityModalProps.create(idField.getText(), showModal, () -> setAnnotationsVisible(false), "ANNOTATIONS", showTabs);
+
+                ReactDOM.render(
+                        React.createElementWithSynapseContext(
+                                SRC.SynapseComponents.EntityModal,
+                                props,
+                                propsProvider.getJsInteropContextProps()
+                        ),
+                        annotationsModalContainer.getElement()
+                );
+            } else {
+                boolean showModal = false;
+                EntityModalProps props =
+                        EntityModalProps.create(idField.getText(), showModal, () -> setAnnotationsVisible(false), "ANNOTATIONS", showTabs);
+                ReactDOM.render(
+                        React.createElementWithSynapseContext(
+                                SRC.SynapseComponents.EntityModal,
+                                props,
+                                propsProvider.getJsInteropContextProps()
+                        ),
+                        annotationsModalContainer.getElement()
+                );
+            }
```

Comment:
If the 'visible' variable changes we have to manually rerender with the changed `showModal` prop. This is super verbose without JSX!